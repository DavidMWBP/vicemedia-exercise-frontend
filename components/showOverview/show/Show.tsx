import {
	StyledEpisodes,
	StyledImage,
	StyledMetadata,
	StyledShow,
	StyledTitle
} from "components/showOverview/show/Show.styled";

export interface Show {
	id: string;
	title: string;
	episodes: number;
	product_image_url: string;
}

function Show({ title, episodes, product_image_url }: Show): JSX.Element {
	return (
		<StyledShow>
			<StyledImage
				src={product_image_url}
				alt={title}
				width={290}
				height={430}
				loading="eager"
			/>
			<StyledMetadata>
				<StyledEpisodes>{episodes} episodes</StyledEpisodes>
				<StyledTitle>{title}</StyledTitle>
			</StyledMetadata>
		</StyledShow>
	);
}

export default Show;
