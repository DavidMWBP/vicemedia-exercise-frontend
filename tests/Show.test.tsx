import Show from "components/showOverview/show/Show";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
	const tree = renderer
		.create(
			<Show
				id="1"
				title="Test Title"
				episodes={20}
				product_image_url="/someimage.png"
			/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
