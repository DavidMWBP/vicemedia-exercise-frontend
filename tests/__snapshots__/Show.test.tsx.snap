// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Showstyled__StyledShow-sc-7uvw6o-0 cGpFxR"
>
  <div
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "margin": 0,
        "maxWidth": "100%",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "maxWidth": "100%",
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        role="presentation"
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjkwIiBoZWlnaHQ9IjQzMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
        style={
          Object {
            "border": "none",
            "display": "block",
            "margin": 0,
            "maxWidth": "100%",
            "padding": 0,
          }
        }
      />
    </div>
    <img
      alt="Test Title"
      className="Showstyled__StyledImage-sc-7uvw6o-1"
      decoding="async"
      src="/_next/image?url=%2Fsomeimage.png&w=640&q=75"
      srcSet="/_next/image?url=%2Fsomeimage.png&w=384&q=75 1x, /_next/image?url=%2Fsomeimage.png&w=640&q=75 2x"
      style={
        Object {
          "border": "none",
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": 0,
          "left": 0,
          "margin": "auto",
          "maxHeight": "100%",
          "maxWidth": "100%",
          "minHeight": "100%",
          "minWidth": "100%",
          "objectFit": undefined,
          "objectPosition": undefined,
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 0,
        }
      }
    />
  </div>
  <div
    className="Showstyled__StyledMetadata-sc-7uvw6o-2 hYvsDa"
  >
    <div
      className="Showstyled__StyledEpisodes-sc-7uvw6o-3 kBHXUu"
    >
      20
       episodes
    </div>
    <h3
      className="Showstyled__StyledTitle-sc-7uvw6o-4 ddSRVy"
    >
      Test Title
    </h3>
  </div>
</div>
`;
